hudi-common/src/main/java/org/apache/hudi/common/util/CollectionUtils.java

```diff
@@ -0,0 +1,132 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.common.util;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class CollectionUtils {
+  /**
+   * Determines whether two iterators contain equal elements in the same order. More specifically,
+   * this method returns {@code true} if {@code iterator1} and {@code iterator2} contain the same
+   * number of elements and every element of {@code iterator1} is equal to the corresponding element
+   * of {@code iterator2}.
+   *
+   * <p>Note that this will modify the supplied iterators, since they will have been advanced some
+   * number of elements forward.
+   */
+  public static boolean elementsEqual(Iterator<?> iterator1, Iterator<?> iterator2) {
+    while (iterator1.hasNext()) {
+      if (!iterator2.hasNext()) {
+        return false;
+      }
+      Object o1 = iterator1.next();
+      Object o2 = iterator2.next();
+      if (!Objects.equals(o1, o2)) {
+        return false;
+      }
+    }
+    return !iterator2.hasNext();
+  }
+
+  @SafeVarargs
+  public static <T> Set<T> createSetFromElements(final T... elements) {
+    return Stream.of(elements).collect(Collectors.toSet());
+  }
+
+  public static <K,V> Map<K, V> createImmutableMap(final K key, final V value) {
+    return Collections.unmodifiableMap(Collections.singletonMap(key, value));
+  }
+
+  @SafeVarargs
+  public static <T> List<T> createImmutableList(final T... elements) {
+    return Collections.unmodifiableList(Stream.of(elements).collect(Collectors.toList()));
+  }
+
+  public static <K,V> Map<K,V> createImmutableMap(final Map<K,V> map) {
+    return Collections.unmodifiableMap(map);
+  }
+
+  @SafeVarargs
+  public static <T> Set<T> createImmutableSet(final T... elements) {
+    return Collections.unmodifiableSet(Stream.of(elements).collect(Collectors.toSet()));
+  }
+
+  public static <T> Set<T> createImmutableSet(final Set<T> set) {
+    return Collections.unmodifiableSet(set);
+  }
+
+  public static <T> List<T> createImmutableList(final List<T> list) {
+    return Collections.unmodifiableList(list);
+  }
+
+  private static Object[] checkElementsNotNull(Object... array) {
+    return checkElementsNotNull(array, array.length);
+  }
+
+  private static Object[] checkElementsNotNull(Object[] array, int length) {
+    for (int i = 0; i < length; i++) {
+      checkElementNotNull(array[i], i);
+    }
+    return array;
+  }
+
+  private static Object checkElementNotNull(Object element, int index) {
+    if (element == null) {
+      throw new NullPointerException("at index " + index);
+    }
+    return element;
+  }
+
+  public static class Maps {
```

Comment:
I am wondering if we can eliminate the need for this class and the builder class, but replacing with plain hashmaps inline? 
- in places where a static HashMap is needed we can use the map initialization syntax I pasted above. 

Don't feel very strongly about this,, but thinking if we can avoid builder patterns where it actually is not used that way.

Reply:
+1 - this was the closest to replacing Guava's ImmutableMap Builder pattern that was being widely used across the project.

Reply:
we can file a follow up JIRA and deal with it later it as well

Reply:
https://issues.apache.org/jira/browse/HUDI-737 filed this

Reply:
Fixed it as part of this PR including removing CollectionUtils.Maps - HUDI-737 can be closed.